File: s3PythonSDK/sample/BucketLoggingSerialTesting.py
from botocore.exceptions import ClientError
from client import client

# test 1. Basic putBucket
#      2. put BucketACL
#      3. put BucketLogging (put log Native & to target bucket)
#      4. get BucketLogging
#      5. Delete Bucket


def main(arg, ownerInfo):
    try:
        for bucket_name in arg:
            client.create_bucket(
                CreateBucketConfiguration={
                    'LocationConstraint': 'ap-southeast-1'
                },
                Bucket=bucket_name,
            )

            client.put_bucket_acl(
                Bucket=bucket_name,
                ACL='log-delivery-write'
            )
        client.put_bucket_logging(
            Bucket=arg[0],
            BucketLoggingStatus={
                'LoggingEnabled': {
                    'TargetBucket': arg[1],
                    'TargetGrants': [
                        {
                            'Grantee': {
                                'ID': ownerInfo[0],
                                'Type': 'CanonicalUser'
                            },
                            'Permission': 'FULL_CONTROL',
                        },
                    ],
                    'TargetPrefix': '',
                },
            },
        )
        #print("Logging List & Give full permission by user ID:")
        #print(repr(client.get_bucket_logging(Bucket=arg[0])['LoggingEnabled']) + "\n")
        #print("Clean up..\n")
        for bucket_name in arg:
            result = client.list_objects_v2(
                Bucket=bucket_name,
            )
            if 'Contents' in result:
                for r in result['Contents']:
                    client.delete_object(
                        Bucket=bucket_name,
                        Key=r['Key']
                    )
            client.delete_bucket(
                Bucket=bucket_name,
            )
        print("Bucket logging Serial test done!\n")
    except ClientError as e:
        print("Error operation : " + e.operation_name)
        print("Error code : " + e.response['Code'])
        print("Error response : " + e.response['Message'])

File: s3PhpSDKExample/sample/BucketLoggingSerialTesting.php
<?php
/*
 * test 1. Basic putBucket
* 		2. put BucketACL
* 		3. put BucketLogging (put log Native & to target bucket)
* 		4. get BucketLogging
* 		5. Delete Bucket
*/

use Aws\S3\Exception\MalformedXMLException;
use Aws\S3\Exception\S3Exception;

require 'client.php';

$bucketname=$argv[1];
$bucketname2=$argv[2];
$ownerCanonicalId=$argv[4];
try {
    echo "\nBucket Logging Serial Testing...\n";
    //CreateBucket
    $client->createBucket(array(
        'Bucket' => $bucketname,
    ));
    $client->createBucket(array(
        'Bucket' => $bucketname2,
    ));
    
    //log_delievery group must have WRITE & READ_ACP Permission
    $acp = [
        
    ];
    
    //SetBucketACL
    $client->putBucketAcl(array(
            'Bucket' => $bucketname,
            'Grants' => [
                [
                    'Grantee' => [
                        'Type' => 'Group',
                        'URI' => 'http://acs.amazonaws.com/groups/global/LogDelivery'
                    ],
                    'Permission' => 'WRITE',
                ],
                [
                    'Grantee' => [
                        'Type' => 'Group',
                        'URI' => 'http://acs.amazonaws.com/groups/global/LogDelivery'
                    ],
                    'Permission' => 'READ_ACP',
                ],
                [
                    'Grantee' => [
                        'Type' => 'CanonicalUser',
                        'ID' => $ownerCanonicalId,
                    ],
                    'Permission' => 'FULL_CONTROL',
                ],
            ],
            'Owner' => [
                'ID' => "{$ownerCanonicalId}"
            ]
    ));
    
    $client->putBucketAcl(array(
            'Bucket' => $bucketname2,
            'Grants' => [
                [
                    'Grantee' => [
                        'Type' => 'Group',
                        'URI' => 'http://acs.amazonaws.com/groups/global/LogDelivery'
                    ],
                    'Permission' => 'WRITE',
                ],
                [
                    'Grantee' => [
                        'Type' => 'Group',
                        'URI' => 'http://acs.amazonaws.com/groups/global/LogDelivery'
                    ],
                    'Permission' => 'READ_ACP',
                ],
                [
                    'Grantee' => [
                        'Type' => 'CanonicalUser',
                        'ID' => $ownerCanonicalId,
                    ],
                    'Permission' => 'FULL_CONTROL',
                ],
            ],
            'Owner' => [
                'ID' => "{$ownerCanonicalId}"
            ]
    ));
    
    //SetBucketLogging
    $client->putBucketLogging(
        array(
            'Bucket' => $bucketname,
            'LoggingEnabled' => array(
                'TargetBucket' => $bucketname2,
                'TargetGrants' => array(
                    [
                        'Grantee' => array(
                            'Type' => 'CanonicalUser',
                            'ID' => $ownerCanonicalId,
                        ),
                        'Permission' => 'FULL_CONTROL',
                    ]
                ),
                'TargetPrefix' => 'log-'
            )
        ),
    );
    
    //Check BucketLogging
    $result=$client->getBucketLogging(array(
        'Bucket' => $bucketname
    ));
    echo json_encode($result['LoggingEnabled']), "\n";
    
    //SetBucketLogging
    $client->putBucketLogging(
        array(
            'Bucket' => $bucketname,
            'BucketLoggingStatus' => [
                'LoggingEnabled' => array(
                    'TargetBucket' => $bucketname,
                    'TargetGrants' => array(
                        [
                            'Grantee' => array(
                                'Type' => 'CanonicalUser',
                                'ID' => $ownerCanonicalId,
                            ),
                            'Permission' => 'FULL_CONTROL',
                        ]
                    ),
                    'TargetPrefix' => 'log-'
                )
            ]
        ),
    );
    
    //Check BucketLogging
    $result=$client->getBucketLogging(array(
            'Bucket' => $bucketname,
    ));
    
    //DeleteBucket
    $client->deleteBucket(array(
            'Bucket' => $bucketname
    ));
    $client->deleteBucket(array(
            'Bucket' => $bucketname2
    ));
} catch (S3Exception $e) {
    echo "Caught an AmazonServiceException.", "\n";
    echo "Error Message:    " . $e->getMessage(). "\n";
    echo "HTTP Status Code: " . $e->getStatusCode(). "\n";
    echo "AWS Error Code:   " . $e->getExceptionCode(). "\n";
    echo "Error Type:       " . $e->getExceptionType(). "\n";
    echo "Request ID:       " . $e->getRequestId(). "\n";
} catch (MalformedXMLException $e) {
    echo $e->__toString();
}
